openfl/component/aggregation_functions/adagrad_adaptive_aggregation.py

```diff
@@ -25,7 +25,7 @@ def __init__(
         *,
         agg_func: AggregationFunction = DEFAULT_AGG_FUNC,
         params: Optional[Dict[str, np.ndarray]] = None,
-        model_interface=None,
+        model_interface: 'ModelInterface' = None, # NOQA
```

Comment:
@ViktoriiaRomanova , why do we have `#NOQA` here? And I guess it should be:
```python3
from openfl.... import ModelInterface
model_interface: Optional[ModelInterface] = None,
```
Do we have [this](https://github.com/intel/openfl/issues/375) issue if we do so?

Reply:
You're right, I need to add `Optional` here, but only like that ` Optional['ModelInterface'] = None, # NOQA`  because of circular import problem, as you mentioned.